src/doc/playback.cpp

```diff
@@ -473,10 +481,96 @@ bool Playback::decrementRepeat(const frame_t frameDelta)
           stop();
           return false;
         }
-        if (newFrame < 0)
-          newFrame = m_sprite->lastFrame();
-        else if (newFrame > m_sprite->lastFrame())
-          newFrame = 0;
+        if (newFrame < 0) {
+          if (m_playing.empty() ||
+              m_playing.back()->tag->fromFrame() != 0 ||
+              m_playing.back()->tag->aniDir() == AniDir::REVERSE) {
+            m_frame = m_sprite->lastFrame();
+            return false;
+          }
```

Comment:
You are right, I was testing this:
With newFrame < 0:
- `m_playing.empty()` never happen, because the only way to have newFrame < 0 is **if we have a tag** in REVERSE or PING PONG REVERSE which frame 0 is contained in that tag.
- `m_playing.back()->tag->fromFrame() != 0` never happens, for the same reason.
- `m_playing.back()->tag->aniDir() == AniDir::REVERSE` never happens, because the the newFrame is calculated in `newFrame = (frameDelta * forward < 0 ? tag->fromFrame()-1: tag->toFrame()+1);` which in reverse cases it **gives newFrame > 1 always**.

Therefore, no condition is needed.

Reply:
Probably I'd put an assert but leave the if condition too (just to avoid a crash in production if there is a future bug).